### PR TITLE
Updated neighbourhood records

### PR DESCRIPTION
**Pull request by stepps00**
This will fix https://github.com/whosonfirst-data/whosonfirst-data/issues/1952.

This PR will include:

- Updated locality records for Jeddah, Riyadh, Mecca, and Damman (and surrounding localities)
- Updated neighbourhood records in those same localities

### CODE REVIEW

**Comment by stepps00** [comment]
**Commit Notes:**

a594ec7: Updates existing neighbourhood geometries, and names. Also adds completely new neighbourhood records to this repo
437d739: Sets `mz:` properties on updated neighbourhood venues
d2ce89f: Corrects a handful of errors made in the first commit
5635a54: Corrects the placetype of a completely new record
181c9db: Flag overlapping, "leftover" neighbourhood points as funky, if they fall within new neighbourhood polygons
58837c1: Updates a handful of locality geometries, centroids, src props

**Comment by stepps00** [comment]
@nvkelso this PR is ready to review together.

File: data/136/102/591/5/1361025915.geojson [hunk 1]
```diff
@@ -27,6 +27,7 @@
     "lbl:max_zoom":18.0,
     "mz:hierarchy_label":1,
     "mz:is_current":1,
+    "mz:is_funky":0,
```
Comment (nvkelso) [inline data/136/102/591/5/1361025915.geojson:30]: Why are we saying it's not funky?

Comment (nvkelso) [reply data/136/102/591/5/1361025915.geojson:30]: There are a few of these

File: data/174/607/202/5/1746072025.geojson [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    "mz:is_current":1,
+    "mz:is_funky":0,
+    "name:ara_x_preferred":[
+        "\u0627\u0644\u0645\u0639\u0630\u0631 \u0627\u0644\u0634\u0645\u0627\u0644\u064a"
+    ],
+    "name:eng_x_preferred":[
+        "Al Maathar Al Shemali"
+    ],
+    "name:eng_x_variant":[
+        "\u0627\u0644\u0645\u0639\u0630\u0631 \u0627\u0644\u0634\u0645\u0627\u0644\u064a"
```
Comment (nvkelso) [inline data/174/607/202/5/1746072025.geojson:26]: This name is Arabic, not an English variant

File: data/174/607/204/7/1746072047.geojson [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    "mz:hierarchy_label":1,
+    "mz:is_current":1,
+    "mz:is_funky":0,
+    "name:ara_x_preferred":[
+        "\u0627\u0644\u0645\u062f\u064a\u0646\u0629 \u0627\u0644\u0635\u0646\u0627\u0639\u064a\u0629 \u0627\u0644\u062b\u0627\u0646\u064a\u0629"
+    ],
+    "name:eng_x_preferred":[
+        "2nd Industrial City"
+    ],
+    "name:eng_x_variant":[
```
Comment (nvkelso) [inline data/174/607/204/7/1746072047.geojson:25]: Ditto on the Arabic wrongly in the English variant name slot.

File: data/174/607/208/9/1746072089.geojson [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    "mz:hierarchy_label":1,
+    "mz:is_current":1,
+    "mz:is_funky":0,
+    "name:ara_x_preferred":[
+        "\u0627\u0644\u0633\u0648\u064a\u062f\u064a \u0627\u0644\u063a\u0631\u0628\u064a"
+    ],
+    "name:eng_x_preferred":[
+        "Suwaidi Al Gharbi"
+    ],
+    "name:eng_x_variant":[
```
Comment (nvkelso) [inline data/174/607/208/9/1746072089.geojson:25]: 3rd example of incorrect Arabic in English alt name slot.

File: data/859/073/99/85907399.geojson [hunk 5]
```diff
@@ -92,7 +92,7 @@
     "wof:lang":[
         "ara"
     ],
-    "wof:lastmodified":1582326292,
+    "wof:lastmodified":1627493447,
```
Comment (nvkelso) [inline data/859/073/99/85907399.geojson:95]: Nit: If nothing about the file is changing (besides the timestamp), then it shouldn't be included in the PR.

File: data/859/074/31/85907431.geojson [hunk 6]
```diff
@@ -4,21 +4,20 @@
...
-    "geom:area":0.0,
+    "geom:area":0.000482,
     "geom:area_square_m":0.0,
-    "geom:bbox":"46.810475,24.675125,46.810475,24.675125",
-    "geom:latitude":24.675125,
-    "geom:longitude":46.810475,
+    "geom:bbox":"46.79613,24.66791,46.827154,24.695813",
+    "geom:latitude":24.681904,
+    "geom:longitude":46.811557,
     "gn:id":"",
```
Comment (nvkelso) [inline data/859/074/31/85907431.geojson:12]: This isn't from the PR, but odd we have a `"gn:id":""` here.

File: data/859/074/65/85907465.geojson [hunk 7]
```diff
@@ -4,21 +4,20 @@
...
+    "lbl:latitude":24.64567,
+    "lbl:longitude":46.721143,
     "lbl:max_zoom":18.0,
-    "mps:latitude":24.649191,
-    "mps:longitude":46.727386,
-    "mz:hierarchy_label":0,
-    "mz:is_current":-1,
+    "mps:latitude":24.64567,
+    "mps:longitude":46.721143,
+    "mz:hierarchy_label":1,
```
Comment (nvkelso) [inline data/859/074/65/85907465.geojson:19]: So all the new neighbourhood geoms are flipping the bit from 0 to 1 for including in PIP results string label construction? That makes sense now they are good polygons.

**Review by nvkelso — COMMENTED** [review]
I left a few comments. Overall looks good. Let's review together via screenshare in QGIS tomorrow.

File: data/136/102/591/5/1361025915.geojson (hunk 1 above)
Comment (stepps00) [reply data/136/102/591/5/1361025915.geojson:30]: I was explicit in calling out some `mz:is_funky` = 0 in this PR, but I can remove these properties (when the value is 0) if they're not useful.

File: data/174/607/202/5/1746072025.geojson (hunk 2 above)
Comment (stepps00) [reply data/174/607/202/5/1746072025.geojson:26]: Ah good catch. I thought I caught all of these cases in a previous commit, but it looks like a few remain. I'll add a new commit to update these.

File: data/859/073/99/85907399.geojson (hunk 5 above)
Comment (stepps00) [reply data/859/073/99/85907399.geojson:95]: I agree.. this must have slipped in during a round of `mz:` property updates. I'll try to root out similar records and reset them so they're not included in this PR's diff.

File: data/859/074/65/85907465.geojson (hunk 7 above)
Comment (stepps00) [reply data/859/074/65/85907465.geojson:19]: Yep, any neighbourhood record with a new polygon will now appear in the label string (hierarchy label = 1).

File: data/859/074/31/85907431.geojson (hunk 6 above)
Comment (stepps00) [reply data/859/074/31/85907431.geojson:12]: Agreed.. should we remove these properties (when empty) in this PR?

File: data/174/607/208/9/1746072089.geojson (hunk 4 above)
Comment (stepps00) [reply data/174/607/208/9/1746072089.geojson]: fixed in 27ea53d

File: data/174/607/204/7/1746072047.geojson (hunk 3 above)
Comment (stepps00) [reply data/174/607/204/7/1746072047.geojson]: fixed in 27ea53d

File: data/174/607/202/5/1746072025.geojson (hunk 2 above)
Comment (stepps00) [reply data/174/607/202/5/1746072025.geojson]: fixed in 27ea53d

File: data/859/073/99/85907399.geojson (hunk 5 above)
Comment (stepps00) [reply data/859/073/99/85907399.geojson]: This record (along with a few dozen others) was reverted to its original state fixed in 67fdf8c.

File: data/859/074/31/85907431.geojson (hunk 6 above)
Comment (nvkelso) [reply data/859/074/31/85907431.geojson:12]: It's probably affecting more tiles, so file a general cleanup issue?

File: data/136/102/591/5/1361025915.geojson (hunk 1 above)
Comment (nvkelso) [reply data/136/102/591/5/1361025915.geojson:30]: It's not wrong. If we want to note it to ourselves that this is _in fact_ not funky then we can keep it. But else general practice should be skipping these.

**Review by nvkelso — APPROVED** [review]
We looked at this together in QGIS, LGTM

**Comment by stepps00** [comment]
A few screenshots of new neighbourhood coverage (green), with locality boundaries (purple), below:

### Riyadh
Screen Shot 2021-07-29 at 4 36 06 PM

### Damman (and surrounding)
Screen Shot 2021-07-29 at 4 36 23 PM

### Medina
Screen Shot 2021-07-29 at 4 35 32 PM

### Jeddah
Screen Shot 2021-07-29 at 4 35 07 PM

**Comment by stepps00** [comment]
I'm going to add a handful of commits to this PR to update records with Point geometries, then pass this off to the PIP tools. I expect this PR to be merged tomorrow.

File: data/859/074/31/85907431.geojson (hunk 6 above)
Comment (stepps00) [reply data/859/074/31/85907431.geojson:12]: Tracking as a subtask in https://github.com/whosonfirst-data/whosonfirst-data/issues/866.

**Comment by stepps00** [comment]
PIP work is complete here. I'll validate the repo, QA a bit more, then merge.
